src/runway/path.py

```diff
@@ -0,0 +1,273 @@
+"""Runway configuration 'path' settings."""
+# pylint: disable=unused-import
+from typing import Dict, List, Optional, Tuple, Union  # noqa: F401
+
+import os
+import logging
+import six
+
+from .sources.git import Git
+
+LOGGER = logging.getLogger('runway')
+
+
+class Path(object):  # pylint: disable=too-many-instance-attributes
+    """Runway configuration ``path`` settings object.
+
+    Path is responsible for parsing the ``path`` property of a Runway
+    configuration. It then can determine if the path specified is
+    locally sourced or remotely sourced through a service such
+    as `Git`_ or S3.
+
+    Local ``path`` variables are defined relative to the root project folder.
+    The value for this cannot be higher than the Runway config file, it must
+    be at the runway file itself or in a sub directory.
+
+    Example:
+        .. code-block:: yaml
+
+            deployments:
+            - modules:
+                - path: my/local/module.cfn
+                - my/local/module.cfn # same as above
+                - ./ # module is in the root
+
+    When the ``path`` is remote, Runway is responsible for fetching
+    the resource and returning the location of it's cached path.
+    The information for retrieving those sources can be controlled via
+    runway rather than manually retrieving each one.
+
+    Example:
+        .. code-block:: yaml
+
+            deployments:
+            - modules:
+                - path: git::git://github.com/your_handle/your_repo.git//my-module.cfn
+
+
+    The ``path`` structure is based on the
+    encoding found in
+    `Terraform modules <https://www.terraform.io/docs/modules/sources.html>`_.
+
+    The values parsed from the string are as follows:
+
+    **source**: Determine if the source is local or remote. The initial
```

Comment:
Would this be better as a header or list item?


`- **source**: Det...`

or you can use `rubric` to create a "header" that does not appear in the toc.

```
    .. rubric:: Source

    Det...
```

Reply:
I tried to do the list item style initially but the formatting gets really strange and all over the place. I tried the rubric header as you suggested and think th is will be a good alternative. Updating.